### PR TITLE
Correct NV12 plane size calcuration

### PR DESCRIPTION
**Pull request by js0701**
Add BO_USE_SCANOUT for NV12 and the correction of bo planse sizes

### CODE REVIEW

File: i915.c [hunk 1]
```diff
@@ -167,6 +166,11 @@ static int i915_add_combinations(struct driver *drv)
 	if (ret)
 		return ret;
 
+	// todo: Chromium require BO_USE_RENDERING but consider to remote it later
+	drv_modify_combination(drv, DRM_FORMAT_YUYV, &metadata, BO_USE_SCANOUT | BO_USE_RENDERING);
```
Comment (kalyankondapally) [inline i915.c]: hmm these are either useful or texturing or scanning out. I dont think we want this combination

File: i915.c [hunk 2]
```diff
@@ -206,40 +210,41 @@ static void get_preferred_cursor_attributes(uint32_t drm_fd,
 }
 
 static int i915_align_dimensions(struct bo *bo, uint32_t tiling, uint32_t *stride,
-				 uint32_t *aligned_height)
+				 uint32_t *aligned_height, uint32_t *horizontal_alignment,
```
Comment (kalyankondapally) [inline i915.c]: Why do we need these changes ?

File: i915.c [hunk 3]
```diff
@@ -351,6 +387,13 @@ static int i915_bo_create(struct bo *bo, uint32_t width, uint32_t height, uint32
 
 	drv_bo_from_format(bo, stride, height, format);
 
+	/*
+	 * drv_bo_from_format does not considerate the alignment for planes
+	 * recalculate the size of planes according to alignment
+	 */
+	if ((h_align > 1) || (v_align > 1))
```
Comment (kalyankondapally) [inline i915.c]: We should probably do this based on formats rather than alignment sizes ?

Comment (js0701) [inline i915.c]: The alignment is achieved from formats.

Comment (kalyankondapally) [inline i915.c]: I think we want to also check for format here and ensure the formats are these planar formats

File: i915.c [hunk 4]
```diff
@@ -262,6 +265,37 @@ static void i915_clflush(void *start, size_t size)
 	}
 }
 
+static void i915_calculate_planessize(struct bo *bo, uint32_t h_align, uint32_t v_align)
+{
+
+	uint32_t offset = 0;
+	uint32_t vertical_subsampling;
```
Comment (kalyankondapally) [inline i915.c]: @xuguangxin ping to also check

**Comment by js0701** [comment]
What is to do next?

**Comment by kalyankondapally** [comment]
Can u rebase

**Comment by kalyankondapally** [comment]
@js0701 Also can u try to get this into chromiumos tree https://chromium.googlesource.com/chromiumos/platform/minigbm/+/d846de667cba72121ec579912d0854729f013731

**Comment by js0701** [comment]
@kalyankondapally  How about we create a new PR I see that many of the changes here (especially cros gralloc )are out dated

**Comment by kalyankondapally** [comment]
@js0701 Closing this, as discussed send it a new PR.

**Pull request closed without merging**